assets/js/googlesitekit/api/cache.js

```diff
@@ -0,0 +1,284 @@
+/**
+ * External dependencies
+ */
+import { isEqual } from 'lodash';
+
+let storageBackend;
+/**
+ * Override the storage backend.
+ *
+ * Largely used for tests. Should not be used directly.
+ *
+ * @param {*} backend Backend to set for the cache.
+ */
+export const _setSelectedStorageBackend = ( backend ) => {
+	storageBackend = backend;
+};
+
+let storageKeyPrefix = 'googlesitekit_';
+/**
+ * Override the key prefix used in storage.
+ *
+ * Largely used for tests.
+ *
+ * @param {string} keyPrefix String to prefix storage keys with.
+ */
+export const _setStorageKeyPrefix = ( keyPrefix ) => {
+	storageKeyPrefix = keyPrefix;
+};
+
+const defaultOrder = [ 'localStorage', 'sessionStorage' ];
+let storageOrder = [ ...defaultOrder	];
+/**
+ * Override the priority of storage mechanisms.
+ *
+ * Largely used for tests. Implicitly resets the selected storage backend,
+ * causing `_getStorage` to re-run its checks for the best available
+ * storage backend.
+ *
+ * @param {Array} order Ordered array of storage backends to use.
+ */
+export const _setStorageOrder = ( order ) => {
+	storageOrder = [ ...order ];
+	_setSelectedStorageBackend( undefined );
+};
+
+/**
+ * Reset the storage mechanism order.
+ *
+ * Largely used for tests. Implicitly resets the selected storage backend,
+ * causing `_getStorage` to re-run its checks for the best available
+ * storage backend.
+ */
+export const _resetDefaultStorageOrder = () => {
+	storageOrder = [ ...defaultOrder ];
+	_setSelectedStorageBackend( undefined );
+};
+
+/**
+ * Detects whether browser storage is both supported and available.
+ *
+ * @param {string} type Browser storage to test. Should be one of `localStorage` or `sessionStorage`.
+ * @return {boolean} True if the given storage is available, false otherwise.
+ */
+export const _isStorageAvailable = async ( type ) => {
+	const storage = global[ type ];
+
+	if ( ! storage ) {
+		return false;
+	}
+
+	try {
+		const x = '__storage_test__';
+
+		storage.setItem( x, x );
+		storage.removeItem( x );
+		return true;
+	} catch ( e ) {
+		return e instanceof DOMException && (
+
+			// everything except Firefox
+			22 === e.code ||
+
+			// Firefox
+			1014 === e.code ||
+
+			// test name field too, because code might not be present
+			// everything except Firefox
+			'QuotaExceededError' === e.name ||
+
+			// Firefox
+			'NS_ERROR_DOM_QUOTA_REACHED' === e.name ) &&
+
+			// acknowledge QuotaExceededError only if there's something already stored
+			0 !== storage.length;
+	}
+};
+
+/**
+ * Gets the storage object to use.
+ *
+ * @return {Object} Return a storage mechanism (`localStorage` or `sessionStorage`) if available; otherwise returns `null`;
+ */
+export const _getStorage = async () => {
```

Comment:
This should not be exported, it should only be used internally in this file. If we need to access it in tests later, we can revisit, but for now there is no need to export this function.